### ISSUE TITLE
Add support for ubuntu 22.04 netplan

### ISSUE THREAD

**Issue by davidzanetti** [hashicorp/terraform-provider-vsphere#1766]
### Community Guidelines

- [X] I have read and agree to the [HashiCorp Community Guidelines ](https://www.hashicorp.com/community-guidelines).
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [ ] Do not leave "+1" or other comments that do not add relevant information or questions.
- [ ] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Description

Latest version of ubuntu (22.04) netplan does not support anymore gateway4 no gateway6.
Instead, it requires :
routes:
  - to: default
    via: <gateway>

### Use Case(s)

Terraform deployment and customization of ubuntu 22.04 templates.

**Comment by tenthirtyam** [comment]
Customization of Ubuntu and later can be performed using either cloud-init and native guest tools.

Are neither of these setting and retaining the expected netplan results? Have you observed a difference between applying customization using the vSphere UI/API and the Terraform provider?

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

**Comment by davidzanetti** [comment]
Goal here is to use terraform and vsphere provider. So, at the end cloudinit.
The generated netplan is :
network:
  version: 2
  renderer: networkd
  ethernets:
    ens160:
	  dhcp4: no
	  dhcp6: no
	  addresses:
	    - 192.168.2.10/24
      gateway4: 192.168.2.254

expected:
network:
  version: 2
  renderer: networkd
  ethernets:
    ens160:
	  dhcp4: no
	  dhcp6: no
	  addresses:
	    - 192.168.2.10/24
	  routes:
	    - to: default
		  via: 192.168.2.254

attached the screen shot of the console with the error.

cloudinit fails as the gateway is not setup. And the gateway is not setup because gateway4 is deprecated.

so, it would be great to have the new schema implemented to the vsphere provider.

![netplan_apply](https://user-images.githubusercontent.com/34554922/195017831-61699c14-4a1d-42c0-aceb-a2f414ef2e55.png)

**Comment by davidzanetti** [comment]
Tested today for ubuntu 20.04 and same problem.

**Comment by tenthirtyam** [comment]
After reviewing `virtual_machine_customize_subresource.go` this appears to be an upstream issue versus provider implementation.

Can perform confirm that the issue does not occur when performing customization natively in vSphere without the provider.

**Comment by tenthirtyam** [comment]
I've performed some testing with the native and cloud-init customization and can confirm that this is an upstream concern and is not specific to the provider. Whilst I close this downstream enhancement, I'll discuss with the concern with the vSphere engineering team.

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

**Comment by tenthirtyam** [comment]
> **Note**
> I've raised the issue with the vSphere engineering team.

Ryan Johnson
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

**Comment by tenthirtyam** [comment]
An update:

netplan command issues a warning message stating that `gateway4` and `gateway6` have been deprecated.

The change on deprecating `gateway4` and `gateway6` was introduced in Canonical's netplan version 0.103:

- netplan Launchpad [Bug](https://bugs.launchpad.net/netplan/+bug/1756590)
- netplan [Pull Request](https://github.com/canonical/netplan/pull/216)
- netplan [Releases](https://github.com/canonical/netplan/release)

In the PR comments, it mentions that _"We do not have a feature flag for this in v0.103, but I've just added the default-routes feature flag for the future. In the meantime gateway4/gateway6 can still be used as-is."_

On the upstream vSphere side, the guest customization scripts and cloud-init are still using these deprecated fields.

1. We have raised an issue to update the customization scripts to stop using these deprecated fields.

2. Our engineers have also submitted an issue on Canonical's cloud-init: 

   - cloud-init LaunchPad [Bug](https://bugs.launchpad.net/cloud-init/+bug/1992512.)

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.
